### DISCUSSION TITLE
Instance-wise metric calculation?

### DISCUSSION THREAD

**Discussion by innat** [open-edge-platform/anomalib#689] [General]
I'm trying to get sample-wise metric score. However, for `good` category samples,  the label and mask are zero. An output of a well-trained model nicely matches the targets. So, computing this with binary metrices for good samples will give zeo scocre!

```python

from torchmetrics.classification import BinaryPrecision

preds  = torch.tensor([0, 0, 0, 0]) # good mask (black pixel)
target = torch.tensor([0, 0, 0, 0]) # predicted mask (black pixel)
metric = BinaryPrecision()
metric(preds, target) # 0
```

In pytorch-lightning, the overal batch output wouold be the **mean** of sample-wise scores, is it correct? For example, the **pixel_scores** (recall, precision, f1) in the following chart.

![image](https://user-images.githubusercontent.com/17668390/200293771-34da8895-032a-4134-9cf1-f3e1776d9f2f.png)
